src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -198,6 +199,24 @@ default Optional<Result> findFirstByParticipationIdAndRatedWithSubmissionOrderBy
             """)
     Optional<Result> findByIdWithEagerFeedbacks(@Param("resultId") long resultId);
 
+    @Query("""
+            SELECT DISTINCT p
+            FROM StudentParticipation p
+                LEFT JOIN FETCH p.results r
+                LEFT JOIN FETCH r.submission s
+                LEFT JOIN FETCH p.submissions
+                LEFT JOIN FETCH r.assessmentNote
+                LEFT JOIN FETCH r.feedbacks f
+                LEFT JOIN FETCH f.testCase
+            WHERE p.exercise.id = :exerciseId
+                AND (
+                    r.id = (SELECT MAX(p_r.id) FROM p.results p_r)
+                    OR r.assessmentType <> de.tum.in.www1.artemis.domain.enumeration.AssessmentType.AUTOMATIC
+                    OR r IS NULL
+                )
+            """)
+    Set<StudentParticipation> findByExerciseIdWithLatestResultAndFeedback(@Param("exerciseId") long exerciseId);
```

Comment:
Why is this method and its `ElseThrow` variant not in the `StudentParticipationRepository`?